velox/functions/remote/utils/restserver/RemoteFunctionRestService.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <boost/asio.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include "velox/common/memory/Memory.h"
+#include "velox/expression/FunctionSignature.h"
+
+namespace facebook::velox::functions {
+
+/// @brief Manages an individual HTTP session.
+/// Handles reading HTTP requests, processing them, and sending responses.
+/// This class re-hosts Velox functions and allows testing their functionality.
+class RestSession : public std::enable_shared_from_this<RestSession> {
```

Comment:
Please mark functions with override if implementing virtual functions of the parent.